include/pybind11/pybind11.h

```diff
@@ -69,6 +69,12 @@ inline bool apply_exception_translators(std::forward_list<ExceptionTranslator> &
     return false;
 }
 
+// Need to use a wrapper function to ensure 1 address
+inline const char *function_capsule_name() {
+    static const char *name = "pybind11_function_capsule";
```

Comment:
Should be constexpr

Reply:
That requires C++23 to support static variables within constexpr functions.

Reply:
Actually, can we use a constexpr function here? I still don't get why this function is necessary. We don't need the address to be the same (just the value)

Reply:
Even if the function is not constexpr, I think that variable can be a `static constexpr const char *`

Reply:
We do actually need the address to be the same to enforce ABI safety. We don't want function_records from different ABI versions interacting.

Reply:
Let's resolve this discussion and keep the ABI talk in the thread below, no need to split it.